control/tests/test_heal.py

```diff
@@ -1,36 +1,150 @@
+import yaml
+
+from django.utils.timezone import utc
 
+from veda_heal import VedaHeal
+from VEDA_OS01.models import Video, Course
 
-class TestEncode(unittest.TestCase):
+
+@ddt
+class HealTests(TestCase):
+    """
+    Tests HEAL process
+    """
 
     def setUp(self):
-        U = URL(
-            videoID=Video.objects.filter(edx_id='XXXXXXXX2014-V00TES1').latest(),
-            encode_profile=Encode.objects.get(product_spec='mobile_low'),
-            encode_url='THIS Is A TEST')
-        U.save()
+        self.heal_instance = VedaHeal()
+        self.auth_yaml = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+            'instance_config.yaml'
+        )
+        self.encode_list = set()
+        with open(self.auth_yaml, 'r') as stream:
+            for key, entry in yaml.load(stream)['encode_dict'].items():
+                for e in entry:
+                    self.encode_list.add(e)
+
+    @data(
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Corrupt File',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Reject',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Hold',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Complete',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': False,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '2',
+            'video_trans_status': 'Ingest',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Corrupt File',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        }
+
+    )
+    @unpack
+    def test_determine_fault(self, edx_id, video_trans_status, video_active, inst_class, video_trans_start):
+        """
+        Tests that determine_fault works as expected.
+        """
+        video_instance = Video(
+            edx_id=edx_id,
+            video_trans_status=video_trans_status,
+            video_trans_start=video_trans_start,
+            video_active=video_active,
+            inst_class=Course()
+        )
+        encode_list = self.heal_instance.determine_fault(video_instance)
```

Comment:
I'm not sure how this will work since we're not saving the video_instance in DB, what will be the result where we're querying from db.

Reply:
@muzaffaryousaf It doesn't query the db - it's just an object import that's fed through the following functions. We _could_ do a database query, but this was sufficient as a passed object.